persistence/postgres-persistence/src/main/java/com/netflix/conductor/postgres/config/outbox/OutboxExecutionDAO.java

```diff
@@ -0,0 +1,258 @@
+import com.netflix.conductor.postgres.dao.PostgresBaseDAO;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+
+@Slf4j
+public class OutboxExecutionDAO extends PostgresBaseDAO implements ExecutionDAO, RateLimitingDAO,
+        PollDataDAO, ConcurrentExecutionLimitDAO {
+
+    private final ExecutionDAO delegate;
+
+    public OutboxExecutionDAO(
+            RetryTemplate retryTemplate,
+            ObjectMapper objectMapper,
+            DataSource dataSource,
+            ExecutionDAO executionDAO) {
+        super(retryTemplate, objectMapper, dataSource);
+        this.delegate = executionDAO;
+        log.info("Outbox table initialized");
+    }
+
+    @Override
+    public List<TaskModel> getPendingTasksByWorkflow(String taskName, String workflowId) {
+        return delegate.getPendingTasksByWorkflow(taskName, workflowId);
+    }
+
+    @Override
+    public List<TaskModel> getTasks(String taskType, String startKey, int count) {
+        return delegate.getTasks(taskType, startKey, count);
+    }
+
+    @Override
+    public List<TaskModel> createTasks(List<TaskModel> tasks) {
+        return delegate.createTasks(tasks);
+    }
+
+    @Override
+    public void updateTask(TaskModel task) {
+        delegate.updateTask(task);
+    }
+
+    @Override
+    @Deprecated
+    public boolean exceedsInProgressLimit(TaskModel task) {
+        return delegate.exceedsInProgressLimit(task);
+    }
+
+    @Override
+    public boolean removeTask(String taskId) {
+        return delegate.removeTask(taskId);
+    }
+
+    @Override
+    public TaskModel getTask(String taskId) {
+        return delegate.getTask(taskId);
+    }
+
+    @Override
+    public List<TaskModel> getTasks(List<String> taskIds) {
+        return delegate.getTasks(taskIds);
+    }
+
+    @Override
+    public List<TaskModel> getPendingTasksForTaskType(String taskType) {
+        return delegate.getPendingTasksForTaskType(taskType);
+    }
+
+    @Override
+    public List<TaskModel> getTasksForWorkflow(String workflowId) {
+        return delegate.getTasksForWorkflow(workflowId);
+    }
+
+    @Override
+    public String createWorkflow(WorkflowModel workflow) {
+        return delegate.createWorkflow(workflow);
+    }
+
+    @Override
+    public String updateWorkflow(WorkflowModel workflow) {
+        String workflowId = delegate.updateWorkflow(workflow);
+        updateOutboxTable(workflow);
+        return workflowId;
+    }
+
+    private void updateOutboxTable(WorkflowModel workflow) {
+        Preconditions.checkNotNull(workflow, "workflow object cannot be null");
+        boolean terminal = workflow.getStatus().isTerminal();
+        try {
+            withTransaction(
+                    tx -> {
+                        if (terminal) writeToOutbox(tx, workflow);
+                    });
+        } catch (NonTransientException e) {
+            log.warn("Writing to outbox table failed");
```

Comment:
This is insufficient. Seeing this message in the log is close to useless. Add the id of workflow to the message and the entire exception.

Reply:
done